public/js/graph/react_graph.js

```diff
@@ -554,6 +556,96 @@ var Graph = React.createClass({
         }
     },
 
+    /**
+     * Calculates the position of the click in relation to the page.
+     * @param {object} e The click event.
+     * @param {HTMLElement} elem The target of the click event.
+     * @return {object} The position of the click.
+     */
+    getClickPosition: function(e, elem) {
+        var parentPosition = getPosition(elem);
+        var xPosition = e.clientX - parentPosition.x;
+        var yPosition = e.clientY - parentPosition.y;
+        xPosition = Math.round(xPosition / 4) * 4; // for snapping!!
+        yPosition = Math.round(yPosition / 4) * 4;
+
+        return { x: xPosition, y: yPosition };
+    },
+
+    /**
+     * Calculates the position of elem in relation to the page.
+     * @param {HTMLElement} elem The target of the click event.
+     * @return {object} The position of elem on the page.
+     */
+    getPosition: function(elem) {
+        var xPosition = 0;
+        var yPosition = 0;
+
+        while (elem) {
+            // || 0 for firefox compatability
+            xPosition += (elem.offsetLeft || 0) - elem.scrollLeft + elem.clientLeft;
+            yPosition += (elem.offsetTop || 0) - elem.scrollTop + elem.clientTop;
+            elem = elem.parentElement; // offsetParent undefined in mozilla
+        }
+        return { x: xPosition, y: yPosition };
+    },
+
+    makeNode: function(xPos, yPos) {
+        var textsList = [];
+        // create text jsons, move to helper function?
```

Comment:
This isn't a bad idea, but because it'll always start as `[]`, I wouldn't worry about it right now.